llvm/lib/Transforms/Scalar/CorrelatedValuePropagation.cpp

```diff
@@ -479,7 +479,8 @@ static bool processAbsIntrinsic(IntrinsicInst *II, LazyValueInfo *LVI) {
 
   // Is X in [0, IntMin]?  NOTE: INT_MIN is fine!
   Result = LVI->getPredicateAt(CmpInst::Predicate::ICMP_ULE, X, IntMin, II,
-                               /*UseBlockValue=*/true);
+                               /*UseBlockValue=*/true,
+                               /*UndefAllowed*/ IsIntMinPoison);
```

Comment:
I'd prefer to switch these to use `getConstantRange()` instead and then do an icmp on the range. Especially as we are doing multiple comparisons on the same range.

This is slightly weaker than getPredicateAt() in some cases, but I don't think it matters here, and it will reduce API changes.

Reply:
This would also allow using getConstantRangeAtUse(), which is probably more valuable anyway.

Reply:
Thanks. Test cases and compile time don't look any worse.
https://llvm-compile-time-tracker.com/compare.php?from=81857940f278e21f7957a2833d4b6ec72819e79f&to=5bd7bbea73315a5f2ae5c73a126b757acac13224&stat=instructions:u